wpiutil/generate_numbers.py

```diff
@@ -0,0 +1,40 @@
+import os
+import shutil
+import sys
+
+MAX_NUM = 20
+
+dirname, _ = os.path.split(os.path.abspath(__file__))
+cmake_binary_dir = sys.argv[1]
+
+with open(f"{dirname}/src/generate/GenericNumber.java.in", "r") as templateFile:
+    template = templateFile.read()
+    rootPath = f"{cmake_binary_dir}/generated/main/java/edu/wpi/first/wpiutil/math/numbers"
+
+    if os.path.exists(rootPath):
+        shutil.rmtree(rootPath)
+    os.makedirs(rootPath)
+
+    for i in range(MAX_NUM + 1):
+        f = open(f"{rootPath}/N{i}.java", "w")
+        f.write(template.replace("${num}", str(i)))
+        f.close()
+
+with open(f"{dirname}/src/generate/Nat.java.in", "r") as templateFile:
+    template = templateFile.read()
+    outputPath = f"{cmake_binary_dir}/generated/main/java/edu/wpi/first/wpiutil/math/Nat.java"
+    getterFile = open(f"{dirname}/src/generate/NatGetter.java.in", "r")
```

Comment:
You can just do
```
with open(f"{dirname}/src/generate/NatGetter.java.in") as getterFile:
    getter = getterFile.read()
```
"with" closes the file automatically and `getter` will be accessible in the outer scope (python doesn't have scope, which is why "with" is a thing). You should be using the "with" idiom for writing as well.